src/Nethermind/Nethermind.Optimism/OptimismReceiptsRootCalculator.cs

```diff
@@ -0,0 +1,44 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Linq;
+using Nethermind.Blockchain.Receipts;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Specs;
+using Nethermind.State.Proofs;
+
+namespace Nethermind.Optimism;
+
+public class OptimismReceiptsRootCalculator : IReceiptsRootCalculator
+{
+    public static readonly OptimismReceiptsRootCalculator Instance = new();
+
+    public Hash256 GetReceiptsRoot(TxReceipt[] receipts, IReceiptSpec spec, Hash256? suggestedRoot)
+    {
+        Hash256 SkipStateAndStatusReceiptsRoot()
+        {
+            receipts.SetSkipStateAndStatusInRlp(true);
+            try
+            {
+                return ReceiptTrie<OptimismTxReceipt>.CalculateRoot(spec, receipts.Cast<OptimismTxReceipt>().ToList(), OptimismReceiptDecoder.Instance);
```

Comment:
I think `CalculateRoot` could take `IEnumerable` and we can avoid `ToList`, this would also make this class same as standard `RootCalculator` only differ by decoder - so we can have one base generic class for them.

Reply:
It was changed to array

Reply:
Ok lets leave it.